src/main/java/core/basesyntax/Ball.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax;
+
+public class Ball {
+    private ColorSupplier.Color color;
```

Comment:
The field 'color' should be of the enum type 'Color' directly, rather than 'ColorSupplier.Color'. If 'Color' is an enum nested inside 'ColorSupplier', it should be made a top-level enum or statically imported.